price-estimator/src/metrics.rs

```diff
@@ -0,0 +1,64 @@
+use anyhow::Result;
+use prometheus::{Histogram, HistogramOpts, HistogramVec, IntCounterVec, Opts, Registry};
+use std::time::Instant;
+use warp::log::Info;
+
+// There are global metrics (response_status and response_time) that are measured through warp's log
+// functionality. The global metrics work on the final response and do not know about the specific
+// route that handled them.
+// And there are metrics (response_time_success) that are labelled per route. We only measures the
+// success response time here as error response times are short and uninteresting and would skew the
+// result.
+
+pub struct Metrics {
+    response_status: IntCounterVec,
+    response_time: Histogram,
+    response_time_success: HistogramVec,
+}
+
+impl Metrics {
+    pub fn new(registry: &Registry) -> Result<Self> {
+        let opts = Opts::new(
+            "price_estimator_response_status",
+            "The status code of a price estimator response.",
+        );
+        let response_status = IntCounterVec::new(opts, &["status"]).unwrap();
+        registry.register(Box::new(response_status.clone()))?;
+
+        let opts = HistogramOpts::new(
+            "price_estimator_response_time_global",
+            "The duration it takes for the price estimator to respond.",
+        );
+        let response_time = Histogram::with_opts(opts).unwrap();
+        registry.register(Box::new(response_time.clone()))?;
+
+        let opts = HistogramOpts::new(
```

Comment:
I'm not a fan of the per route handling, but if we keep it can we rename the metric/methods to make sure it's response_time_per_route? It's well described in the comments but I'd prefer if the variable and metrics were more self explanatory.